spidermon/contrib/scrapy/monitors/suites.py

```diff
@@ -38,6 +38,30 @@ class SpiderCloseMonitorSuite(MonitorSuite):
             )
     """
 
+    def __init__(
+        self,
+        name=None,
+        monitors=None,
+        monitors_finished_actions=None,
+        monitors_passed_actions=None,
+        monitors_failed_actions=None,
+        order=None,
+        crawler=None,
+    ):
+        super().__init__(
+            name,
+            monitors,
+            monitors_finished_actions,
+            monitors_passed_actions,
+            monitors_failed_actions,
+            order,
+            crawler,
+        )
+        if dict(crawler.settings).get("SPIDERMON_MONITOR_SKIPPING_RULES"):
+            skip_rules = crawler.settings.get("SPIDERMON_MONITOR_SKIPPING_RULES")
+            for monitor in self.monitors:
+                monitor.skip_rules = skip_rules
```

Comment:
Why are we assigning the dict of monitors to every monitor. Since rules are based on monitor name, couldn’t we filter them at this point, and only assign to each monitor its own rules, and not set rules altogether on monitors for which there are no rules?

Reply:
At this point, monitor's properties are not set (we can't get the monitor.name or monitor.monitor_name)

Reply:
I see :sweat:.

I was hoping monitor had access to the Scrapy crawler of settings, but I see that’ implementing that would me a major change, and I don’t think it makes sense to make it as part of this.

Given the current code base, I think your approach is the cleanest it can be. Great work!